### PR TITLE
Fix of #347, #324, #312 and #323 

### PR DESCRIPTION
**Pull request by pmk65**
Fix of #347, #324, #312 and #323

### CODE REVIEW

File: src/validator.js [hunk 1]
```diff
@@ -537,19 +537,19 @@ JSONEditor.Validator = Class.extend({
...
     if(['date', 'time', 'datetime-local'].indexOf(schema.format) != -1) {
 
-      var validator = {
-          'date': /^(\d{4}\D\d{2}\D\d{2})?$/,
-          'time': /^(\d{2}:\d{2}(?::\d{2})?)?$/,
-          'datetime-local': /^(\d{4}\D\d{2}\D\d{2} \d{2}:\d{2}(?::\d{2})?)?$/
+      var validatorRx = {
+        'date': /^(\d{4}\D\d{2}\D\d{2})?$/,
+        'time': /^(\d{2}:\d{2}(?::\d{2})?)?$/,
+        'datetime-local': /^(\d{4}\D\d{2}\D\d{2}[ T]\d{2}:\d{2}(?::\d{2})?)?$/
```
Comment (schmunk42) [inline src/validator.js]: Is it on purpose that you allow a ` `(space) in the regex?

Comment (pmk65) [reply src/validator.js]: Yes. Chrome returns the date string differently from Firefox and others. So it has to be able to validate with either space or "T"

Comment (schmunk42) [reply src/validator.js]: Do we need to take care about the different browsers regarding reformatting the values?
Would people using FF or Chrome save different values to a database? Or would this have to be done on the server?

Comment (pmk65) [reply src/validator.js]: Storing dates as strings in databases is generally not a good idea. A integer epoch date (unix timestamp) is much better.  
The string format returned is based on the format selected (flatpickr) or the default browser return value.

**Comment by schmunk42** [comment]
Thank you!

**Pull request merged**